src/hooks/useForm/FormProvider.js

```diff
@@ -146,18 +149,18 @@ function FormProvider({validate, shouldValidateOnBlur, shouldValidateOnChange, c
             const newRef = propsToParse.ref || createRef();
             inputRefs[inputID] = newRef;
 
-            // We want to initialize the input value if it's undefined
-            if (_.isUndefined(inputValues[inputID])) {
-                inputValues[inputID] = propsToParse.defaultValue || getInitialValueByType(propsToParse.valueType);
+            if (!_.isUndefined(propsToParse.value)) {
+                inputValues[inputID] = propsToParse.value;
             }
 
             // We force the form to set the input value from the defaultValue props if there is a saved valid value
-            if (propsToParse.shouldUseDefaultValue) {
+            else if (propsToParse.shouldUseDefaultValue) {
```

Comment:
This `else if` style looks a bit odd and different than what we usually have in the codebase. Could we remove the extra lines so it's in the same like as the closing bracket of the `if` statement and we move the comments inside the block?